src/common/mfu_daos.c

```diff
@@ -121,118 +104,285 @@ static bool daos_check_prefix(char* path, const char* dfs_prefix, char** suffix)
     if (strncmp(path, dfs_prefix, prefix_len) == 0) {
         /* if equal, assume root */
         if (path_len == prefix_len) {
-            *suffix = "/";
+            strcpy(suffix, "/");
             is_prefix = true;
         }
         /* if path is longer, it must start with '/' */
-        else if (path_len > prefix_len &&
-            path[prefix_len] == '/') {
-            *suffix = path + prefix_len;
+        else if (path_len > prefix_len && path[prefix_len] == '/') {
+            strcpy(suffix, path + prefix_len);
             is_prefix = true;
         }
     }
     return is_prefix;
 }
 
+/*
+ * Parse a path of the format:
+ * daos://<pool>/<cont>/<path> | <UNS path> | <POSIX path>
+ * Modifies "path" to be the relative container path, defaulting to "/".
+ * Returns 0 IFF a daos path was successfully parsed.
+ */
+static int daos_parse_path(
+    char* path,
+    uuid_t* p_uuid,
+    uuid_t* c_uuid)
+{
+    struct duns_attr_t  dattr = {0};
+    size_t              path_len = strlen(path);
+    int                 rc;
+
+    rc = duns_resolve_path(path, &dattr);
+    if (rc == 0) {
```

Comment:
@daltonbohning if the user is not using a UNS path, wouldn't duns_resolve_path always return a non-zero rc? If you are using the rel_path passed back by dattr then I guess I am confused how you are retrieving the rel_path for non-UNS paths.

Reply:
@dsikich You're right, `rc != 0` for "regular POSIX" paths. In that case, just don't do anything. Ignore `dattr` entirely for `rc != 0`, don't modify the path, don't use `rel_path`, don't set pool/cont uuids, etc. Then, this "POSIX" path will be handled normally by `dcp` and and the other internal functions.

The way I'm using `duns_resolve_path` here is basically as an "interceptor". I call `duns_resolve_path` on the path passed in. This means that `rc == 0` for daos:// and UNS paths, in which case we DO use rel_path and set the pool/cont uuids.

An improvement that could be made here is something like:
```
if (rc != 0 && strcnmp(path, "daos://", 8) == 0) {
    // the user explicitly specified a daos path, but it was somehow invalid
    // do NOT treat this as a POSIX path
}
```
I will add this.

Reply:
@daltonbohning ok, I see. I was looking at how you parsed the paths here, because I don't think my other PR for the DAOS filesystem copy is going to get through unless I update using this function as well (instead of all the manual parsing I'm using). I am planning to use how you re-worked it here. It should simplify and remove a lot of the comments made on the PR I'm trying to get through.

Reply:
@dsikich Sounds good! There are definitely some more improvements that can be made here with this parsing, but I think this is at least stable. My latest commit also includes another error check.